src/test/test.cpp

```diff
@@ -495,6 +519,11 @@ for_versions(std::vector<uint32> const& versions, Application& app,
         {
             continue;
         }
+        if (gTestingVersion != v)
+        {
+            continue;
+        }
+
         SECTION("protocol version " + std::to_string(v))
         {
             {
```

Comment:
Done. Removing this required updating some tests to stop using 'USE_CONFIG_FOR_GENESIS=false'. I had to make a significant change to the inflation tests though. I pulled a subset of the inflation tests into another test called `inflation total coins`. 

The issue here is that we previously did the setup in V0, and then updated the header for each version. I initially updated the test so 'USE_CONFIG_FOR_GENESIS==true', but this didn't work because the test is written with the assumption that CAP-0005 isn't enabled, which is false for V11. I ended up just pulling the test out and mimicking the old behavior. The other option I thought about was just removing this test for V11 since inflation is disabled and we're past V11, but that didn't sound like the safest option.

Reply:
Also, adding a check here to make sure we're on the right version resulted in a test failure in the Upgrade tests, so I didn't add it, but I'll look into fixing those failures.

Reply:
The last commit adds a check for the ledger version.